addons/t9n/models/resource.py

```diff
@@ -14,3 +19,52 @@ class Resource(models.Model):
     project_id = fields.Many2one(
         comodel_name="t9n.project",
     )
+
+    _sql_constraints = [
+        (
+            "file_name_unique",
+            "unique(file_name)",
+            "A file with the same name already exists!"
+        ),
+    ]
+
+    def _decode_resource_file(self, resource_file):
+        try:
+            file_content = base64.b64decode(resource_file.get("file")).decode()
+            po_obj = polib.pofile(file_content)
+        except (IOError, UnicodeDecodeError):
+            raise UserError(_("This file: %s should be a .po file with a valid syntax." % resource_file.get("file_name")))
```

Comment:
I would push the error to a list, finish reading all the files, and then raise an error with the list of all the problematic files at once, so that the user can fix everything in a single pass